ci-operator/step-registry/baremetalds/assisted/operator/setup/baremetalds-assisted-operator-setup-commands.sh

```diff
@@ -34,25 +36,52 @@ cd "\${REPO_DIR}"
 
 export DISCONNECTED="${DISCONNECTED:-}"
 
+if [ "\${DISCONNECTED}" = "true" ]; then
+  export LOCAL_REGISTRY="\${LOCAL_REGISTRY_DNS_NAME}:\${LOCAL_REGISTRY_PORT}"
+
+  source deploy/operator/mirror_utils.sh
+
+  export AUTHFILE="\${XDG_RUNTIME_DIR}/containers/auth.json"
+  mkdir -p \$(dirname \${AUTHFILE})
+
+  merge_authfiles "\${PULL_SECRET_FILE}" "\${REGISTRY_CREDS}" "\${AUTHFILE}"
+fi
+
 echo "### Setup hive..."
 
 if [ "\${DISCONNECTED}" = "true" ]; then
   hack/setup_env.sh hive_from_upstream
-
-  export LOCAL_REGISTRY="\${LOCAL_REGISTRY_DNS_NAME}:\${LOCAL_REGISTRY_PORT}"
-  export AUTHFILE="\${REGISTRY_CREDS}"
   deploy/operator/setup_hive.sh from_upstream
```

Comment:
Could we make `setup_hive.sh` smarter to identify when `AUTHFILE` and `LOCAL_REGISTRY` are used and call the matching function?

Reply:
scoping doesn't make sense I afraid
the action is either ``from_upstream/with_olm`` and those are only parameters